### PR TITLE
Removes access restrictions from post-roundstart RnD Consoles (kind of)

### PR DESCRIPTION
**Pull request by uraniummeltdown**
Replaces the Core RnD Console board in Tech Storage with the Public RnD Console board, which is identical but does not have access restrictions. Made the required access for changing RnD Console boards, Scientist instead of RD. New RnD Consoles start off with no upload/download on the Server, this must be enabled by RD/AI.

:cl:
tweak:Removes Core RnD Console board in Tech Storage, adds Public RnD Console board in its place.
tweak:RnD Console boards can now be changed by swiping with IDs holding Scientist access or above.
:cl:

### CODE REVIEW

**Comment by IcyV** [comment]
Not entirely sure how I feel on this idea as a whole but I'm all for the RD being able to choose its access level. I think this might be better by leaving the board as is but introducing the access changes. Let's an RD choose how the machine could potentially be used or opens it to having their ID stolen and letting other people choose.

**Comment by Crazylemon64** [comment]
@IcyV The RD always could configure the access on the R&D console board, that's been around even before this PR

**Comment by IcyV** [comment]
@Crazylemon64 Huh. Can't believe I never noticed that. Not that I especially play RD over scientist often.

**Comment by uraniummeltdown** [comment]
>If this is going to be done, I'd just do away with the access requirement altogether; don't keep the whole public/private access switching thing, just kill it off.

I'd keep that as it is still useful, the RnD console in Science will always have the most ores in it (if Scientists don't neglect RnD to blow themselves up) and therefore is the most capable of printing high material cost items, having an access lock there is good IMO. I see no reason to remove it.

**Comment by Lady-Luck** [comment]
I am very much against access locks that don't take place within the interface of the computers.

Just clicking on a console and it's just 'access denied' is lame.

**Comment by GeneralChaos81** [comment]
If it's public level access to a high level machine, the board should go in secure tech storage.

**Comment by Fox-McCloud** [comment]
If everything is essentially going to be public access for this, there's not much point to keeping the R&D lab one locked, specifically, especially when you can just grab the one from tech storage.

Again, if this is going to be done, then just remove the access restrictions on all the consoles.

**Comment by Regen1** [comment]
Fuck no

This is a horrible change

RND is already the main salt shaker on the codebase, and now you want to make it available to everyone? 

If this should have any remote chance of my approval, rework the RnD server so newly created consoles arent given access to the server, and the RD has to give it to them

**Comment by uraniummeltdown** [comment]
>If it's public level access to a high level machine, the board should go in secure tech storage.

I don't feel it should go in secure tech storage as it's not really capable of turning shifts around like the other boards in there (AI Upload/Cyborg Upload for subverting/resetting silicons; Comms Console for calling/recalling shuttle, ERT or contacting Syndicate; ID Computer for access level editing; Exosuit and Robotics control consoles for blowing borgs/mechs)

>If everything is essentially going to be public access for this, there's not much point to keeping the R&D lab one locked, specifically, especially when you can just grab the one from tech storage.

>Again, if this is going to be done, then just remove the access restrictions on all the consoles.

The RnD Lab one should be locked as it's the main one, with these changes even though the Mime is now able to set up RnD in their office provided they obtain parts and boards, they are still not responsible for doing RnD and upgrading the station as that is the job of Scientists. In most shifts the RnD Lab one will still be the place where all the magic happens and most of the ores go. There is no harm in leaving the access restrictions in.

Also I will be making some changes.

**Comment by uraniummeltdown** [comment]
So now the Circuit Imprinter still prints RnD Core boards by default, however Scientists and up can now change the access on the boards.
Also the only Public RnD board at roundstart exists in Tech Storage, which I think should stay.

**Comment by uraniummeltdown** [comment]
I just tested it and newly built RnD consoles (no matter what board you use) start off without upload/download access on the RnD Server, you will need the RD or AI to add your console to it, the RD can also build another RnD Server for post roundstart consoles where he can delete certain designs (for example deleting Advanced Energy Guns, Durand parts, etc.) so they do not get added even when syncing new consoles.

**Comment by Necaladun** [comment]
Very unsure of this. Giving so much access to high level toys is a big change, and also needs clarification in SoP/Space Law. Will Engineers be allowed to go make them?

I'm kinda +1 on the idea of taking the monopoly away from R&D so much, but obviously against such a powerful thing for so many people.

**Comment by uraniummeltdown** [comment]
>Very unsure of this. Giving so much access to high level toys is a big change, and also needs clarification in SoP/Space Law. Will Engineers be allowed to go make them?

>I'm kinda +1 on the idea of taking the monopoly away from R&D so much, but obviously against such a powerful thing for so many people.

Even with maximized research levels, there aren't many dangerous items you can produce with RnD. New consoles start off unsynced from the Server, so you must do RnD yourself to increase tech levels on your console or ask the RD/AI to add your console to the Server. The Server also displays the location of RnD consoles, so nobody will be going around with a hidden lab in maint. The RD can also remove certain technologies or designs from a server, if they move all roundstart RnD consoles to one server and all built ones to the other (Server room starts with 2 servers), they can easily remove the questionable items, so it won't be as easy as "build computer and Protolathe, press button, get xray lasers and combat implants".

I believe with this change the RD will become more of an actual Research Director, being able to manage Research across the station.

As for SOP and Space Law, existing Science SOP can be applied to anyone with a console, so the Engineers won't be able to build themselves Gygaxes and give the excuse "Ah but I'm not a Roboticist you see so it's allowed" and the QM can't build an AI or AI Upload and say "Shitcurity I'm not a Scientist!" as he gets dragged off to the slammer.

**Comment by Necaladun** [comment]
Yeah I really like this for giving the RD actually...management to do.

If it's unsynced by default, can be tracked, and everything like that, then I like it - just will need to update SoP/space law to make it clear who is and isn't allowed to build them or with what permission. Not a matter for the PR though!

File: _maps/map_files/cyberiad/cyberiad.dmm [hunk 1]
```diff
@@ -5993,7 +5993,7 @@
...
 "clo" = (/obj/structure/table,/obj/item/device/aicard,/obj/item/weapon/aiModule/reset,/turf/simulated/floor/plating,/area/storage/tech)
-"clp" = (/obj/structure/rack{dir = 8; layer = 2.9},/obj/item/weapon/circuitboard/rdconsole{pixel_x = -4},/obj/item/weapon/circuitboard/rdserver{pixel_x = 4; pixel_y = -2},/obj/item/weapon/circuitboard/destructive_analyzer,/obj/item/weapon/circuitboard/protolathe{pixel_x = -2; pixel_y = 3},/obj/structure/cable{d1 = 1; d2 = 2; icon_state = "1-2"; tag = ""},/obj/item/weapon/circuitboard/circuit_imprinter{pixel_x = -4; pixel_y = -3},/turf/simulated/floor/plating,/area/storage/tech)
+"clp" = (/obj/structure/rack{dir = 8; layer = 2.9},/obj/item/weapon/circuitboard/rdserver{pixel_x = 4; pixel_y = -2},/obj/item/weapon/circuitboard/destructive_analyzer,/obj/item/weapon/circuitboard/protolathe{pixel_x = -2; pixel_y = 3},/obj/structure/cable{d1 = 1; d2 = 2; icon_state = "1-2"; tag = ""},/obj/item/weapon/circuitboard/circuit_imprinter{pixel_x = -4; pixel_y = -3},/obj/item/weapon/circuitboard/rdconsole/public,/turf/simulated/floor/plating,/area/storage/tech)
```
Comment (Fox-McCloud) [inline _maps/map_files/cyberiad/cyberiad.dmm]: should have replaced the original circuitboard with this, not moved it.

File: code/modules/research/designs/comp_board_designs.dm [hunk 2]
```diff
@@ -214,7 +214,7 @@
 
 /datum/design/rdconsole
 	name = "Console Board (R&D Console)"
-	desc = "Allows for the construction of circuit boards used to build a new R&D console."
+	desc = "Allows for the construction of circuit boards used to build a new R&D console. Can be swiped with a Scientist level ID to manage access levels."
```
Comment (Crazylemon64) [inline code/modules/research/designs/comp_board_designs.dm]: I'm not sure this description is ever visible, but change it as you please.

**Comment by Fox-McCloud** [comment]
A step in the right direction, I suppose. Still think the access should just be axed, altogether at this stage.

**Pull request merged**